### PR TITLE
Add checkin functionality

### PR DESCRIPTION
**Pull request by masnoble**
pull and test around to make sure that's what you want. 

this seemed to be the best thing that I could make from what we have currently. I tried to leave it improveable

NOTE: make sure you delete your cookies if you relaunch your local server. You need fresh cookies for the report generator and the token seems to time out each time the localhost is restarted

### CODE REVIEW

File: components/CheckinCard.tsx [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
-  )
+import { Registration, Student } from "@prisma/client";
+import { useState } from "react";
+
+
+
+export default function CheckinCard({ student, registration }: { student: Student, registration: Registration }) {
+    const [canCheckIn, setCanCheckIn] = useState(true)
+
+    const checkIn = async (regId: String) => {
```
Comment (SpBills) [inline components/CheckinCard.tsx:11]: this is fine from an MVP perspective. instead, we would rather have something like...

```ts
await fetch(`/api/checkin/${regId}/register`, {
    method: "PUT",
})
```

from a RESTful POV.

File: pages/idcheckin.tsx [hunk 2]
```diff
@@ -53,13 +60,15 @@ export default function IdCheckIn() {
                     <TextField value={idFormData} onChange={e => setIdFormData(e.target.value)} label="5 Digit Student ID" name="id" variant="standard" sx={{ width: "100%", mt: 1 }} inputRef={input => input && input.focus()} />
                 </StyledForm>
             </Box>
-            {student ?
-                <CheckinCard student={student} checkIn={() => alert("User has checked in.")} /> :
+            {
+                student && registrations[0] ?
```
Comment (SpBills) [inline pages/idcheckin.tsx:64]: prefer `registrations.length !== 0`

**Review by SpBills — CHANGES_REQUESTED** [review]
Not implemented on `namecheckin`. I tried to see if I could do it with some simple changes but it's a bit larger than I have time for at the moment. Let me know what you have the capacity for.

**Comment by masnoble** [comment]
Forced with lease to rebase on main so that there's no merge conflicts

File: pages/namecheckin.tsx [hunk 3]
```diff
@@ -49,6 +50,15 @@ export default function NameCheckIn() {
         const response = await fetch(endpoint);
         const studentsInfo = await response.json();
         setStudents(studentsInfo);
+        
+        
+        for(const std of studentsInfo) {
```
Comment (SpBills) [inline pages/namecheckin.tsx:55]: prefer map syntax here. but also, is `regs` an array?

```ts
setRegistrations(await Promise.all( studentsInfo.map(s => {
  const regEndpoint = `/api/register?id=${std.id}&evt=${event}`;
  const regResponse = await fetch(regEndpoint);
  const regs = await regResponse.json();

  return regs;
}) ));
```

**Review by SpBills — APPROVED** [review]
Kinda nitpicky. LGTM tho.

File: pages/namecheckin.tsx (hunk 3 above)
Comment (masnoble) [reply pages/namecheckin.tsx:55]: yes regs is an array. We're finding all registrations for that user. It may be 1, or it may be a bunch

**Pull request merged**